assets/src/edit-story/components/mediaPicker/useMediaPicker.js

```diff
@@ -153,5 +156,117 @@ export default function useMediaPicker({
     ]
   );
 
-  return openMediaPicker;
+  const openCropper = useCallback(
+    (evt) => {
+      trackEvent('open_media_crop_modal');
+      // If a user does not have the rights to upload to the media library, do not show the media picker.
+      if (!hasUploadMediaAction) {
+        if (onPermissionError) {
+          onPermissionError();
+        }
+        evt.preventDefault();
+        return;
+      }
+
+      const control = {
+        id: 'control-id',
+        params,
+        mustBeCropped,
+      };
+
+      // Create the media frame.
+      const fileFrame = global.wp.media({
+        button: {
+          text: buttonInsertText,
+          close: false,
+        },
+        states: [
+          new wp.media.controller.Library({
+            title,
+            library: wp.media.query({ type }),
+            button: {
+              text: buttonInsertText,
+              close: false,
+            },
+            multiple,
+            suggestedWidth: params.width,
+            suggestedHeight: params.height,
+          }),
+          new wp.media.controller.CustomizeImageCropper({
+            imgSelectOptions: calculateImageSelectOptions,
+            control,
+          }),
+        ],
+      });
+
+      fileFrame.on('cropped', (attachment) => {
+        updateMedia(attachment.id, { media_source: 'editor' });
+        onSelect(attachment);
+      });
+
+      fileFrame.on('skippedcrop', onSelect);
+
+      fileFrame.on('select', () => {
+        const mediaPickerEl = fileFrame
+          .state()
+          .get('selection')
+          .first()
+          .toJSON();
+
+        // Only allow user to select a mime type from allowed list.
+        if (Array.isArray(type) && !type.includes(mediaPickerEl.mime)) {
+          fileFrame.close();
+          showSnackbar({ message: onSelectErrorMessage });
+
+          return;
+        }
+
+        if (
+          control.params.width === mediaPickerEl.width &&
+          control.params.height === mediaPickerEl.height &&
+          !control.params.flex_width &&
+          !control.params.flex_height
```

Comment:
We don't use `flex_width` and `flex_height` anywhere / set these to `false` anyway. We might just as well remove all flex* logic to keep things simpler.